src/python/pants/engine/parser.py

```diff
@@ -24,7 +24,7 @@ def table(self):
   def constraint(self):
     """Returns the typeconstraint for the symbol table"""
     # NB Sort types so that multiple calls get the same tuples.
-    symbol_table_types = sorted(set(self.table().values()))
+    symbol_table_types = sorted(set(self.table().values()), key=hash)
```

Comment:
Is it possible to sort these by "fully qualified string name" or something? I'm worried that the hash will not be stable across hosts (maybe it already wasn't).